package/MDAnalysis/core/groups.py

```diff
@@ -725,6 +726,20 @@ def dimensions(self):
     def dimensions(self, dimensions):
         self.universe.trajectory.ts.dimensions = dimensions
 
+    @staticmethod
+    def _create_unique_group(_unique):
+        # Since we know that _unique is a unique AtomGroup, we set its
+        # uniqueness caches from here:
+        _unique._cache['isunique'] = True
+        _unique._cache['unique'] = _unique
+        return _unique
+
+    @property
+    @cached('issorted')
+    def issorted(self):
```

Comment:
I think this can be done without sorting. Also, ``np.argmin`` has a lazy implementation that short-circuits for boolean arrays (in contrast to ``np.all``). So the following should be faster, especially for unsorted arrays (not thoroughly tested, though):
```python
def issorted(self):
    mask = self.ix[:-1] <= self.ix[1:]
    try:
        return mask[0] and mask.argmin() == 0
    except IndexError:
        # Empty atom groups are sorted, I guess...
        return True
```

Reply:
Ah, clever, thank you! If anyone else was wondering whether `np.any` short-circuits with an inverted mask: oddly, apparently not.

```python
>>> x = np.zeros(100000, dtype=bool)
>>> timeit.timeit(x.any)
3.772446749999972
>>> timeit.timeit(lambda: x[x.argmax()])
2.405966604000241
>>> x[1] = True
>>> timeit.timeit(x.any)
3.6103378889997657
>>> timeit.timeit(lambda: x[x.argmax()])
0.4463869180003712
```